packages/arkor-local/python/mlx/dataset_prep.py

```
@@ -0,0 +1,449 @@
+        old_scheme = urllib.parse.urlparse(req.full_url).scheme
+        new_scheme = urllib.parse.urlparse(newurl).scheme
+        if old_scheme == "https" and new_scheme != "https":
+            # Refused outright, not merely stripped of its Authorization
+            # header: the rows themselves would arrive over plaintext, and
+            # a network attacker who can rewrite them poisons the adapter.
+            # A redirect can also carry pre-signed credentials in the new
+            # URL's query, which header stripping does not protect.
+            raise DatasetPrepError(
+                "blob dataset redirect downgrades https to "
+                f"{new_scheme!r}; refusing to follow it"
+            )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)